src/main/java/dev/codesupport/web/api/service/ArticleServiceImpl.java

```diff
@@ -0,0 +1,225 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class ArticleServiceImpl implements ArticleService {
+
+    private final CrudOperations<ArticleEntity, Article, Long> crudOperations;
+    private final PublishedArticleRepository publishedArticleRepository;
+    private final ArticleRepository articleRepository;
+
+    @Autowired
+    public ArticleServiceImpl(
+            PublishedArticleRepository publishedArticleRepository,
+            ArticleRepository articleRepository,
+            TagSetToTagsRepository tagSetToTagsRepository,
+            TagSetRepository tagSetRepository,
+            TagRepository tagRepository
+    ) {
+        crudOperations = new CrudOperations<>(articleRepository, ArticleEntity.class, Article.class);
+        CrudLogic crudLogic = new CrudLogic(articleRepository, tagRepository, tagSetRepository, tagSetToTagsRepository);
+        crudOperations.setPreSaveEntities(crudLogic.preSaveLogic());
+        crudOperations.setPreGetEntities(crudLogic.preGetLogic());
+        this.publishedArticleRepository = publishedArticleRepository;
+        this.articleRepository = articleRepository;
+    }
+
+    @Override
+    public List<Article> findAllArticles() {
+        return crudOperations.getAll();
+    }
+
+    @Override
+    public Article getArticleById(Long id) {
+        return crudOperations.getById(id);
+    }
+
+    @Override
+    public Article createArticle(Article article) {
+        return crudOperations.createEntity(article);
+    }
+
+    @Override
+    public Article updateArticle(Article article) {
+        return null;
+    }
+
+    @Override
+    public Article deleteArticle(Article article) {
+        return null;
+    }
+
+    @Override
+    public Article publishArticle(Article article) {
+        Optional<ArticleEntity> optionalArticle = articleRepository.findById(article.getId());
+
+        if (optionalArticle.isPresent()) {
+            PublishedArticleEntity publishedArticleEntity;
+            Optional<PublishedArticleEntity> optionalPA = publishedArticleRepository.findByArticleCode(optionalArticle.get().getArticleCode());
+
+            if (optionalPA.isPresent()) {
+                publishedArticleEntity = updatePublishedArticle(optionalPA.get(), optionalArticle.get());
+            } else {
+                publishedArticleEntity = createPublishedArticle(optionalArticle.get());
+            }
+
+            publishedArticleRepository.save(publishedArticleEntity);
+        } else {
+             throw new ResourceNotFoundException(ResourceNotFoundException.Reason.NOT_FOUND);
+        }
+        return null;
+    }
+
+    PublishedArticleEntity updatePublishedArticle(PublishedArticleEntity paEntity, ArticleEntity article) {
+        paEntity.setArticleId(article.getId());
+        return paEntity;
+    }
+
+    PublishedArticleEntity createPublishedArticle(ArticleEntity article) {
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/codesupport/website-backend/pullRequest?prid=6300789)